### PR TITLE
feat: extra pixel highlighter

### PR DESCRIPTION
**Pull request by kateberryd**
-  issue #100 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ function App() {
     const colorIdx = color;
     const colorHex = `#${colors[colorIdx]}FF`;
     context.fillStyle = colorHex;
-    context.fillRect(x, y, 1, 1);
+    context.fillRect(x, y, 10, 10);
```
Comment (b-j-roberts) [inline frontend/src/App.js:132]: This should still be (1, 1), why was it changed?

File: frontend/src/App.js [hunk 2]
```diff
@@ -143,7 +143,7 @@ function App() {
   const [basePixelUp, setBasePixelUp] = useState(false);
   const [factionPixelsData, setFactionPixelsData] = useState([]);
   const [factionPixels, setFactionPixels] = useState([]);
-  const [extraPixels, setExtraPixels] = useState(0);
+  const [extraPixels, setExtraPixels] = useState(4);
```
Comment (b-j-roberts) [inline frontend/src/App.js:146]: Please keep this 0 after testing

File: frontend/src/App.js [hunk 3]
```diff
@@ -302,7 +302,7 @@ function App() {
       const pixel = extraPixelsData[index];
       const x = pixel.x;
       const y = pixel.y;
-      context.clearRect(x, y, 1, 1);
+      context.clearRect(x, y, 4, 4);
```
Comment (b-j-roberts) [inline frontend/src/App.js:305]: This also should still be 1, 1 I think

File: frontend/src/canvas/CanvasContainer.js [hunk 4]
```diff
@@ -259,6 +259,17 @@ const CanvasContainer = (props) => {
...
       return null;
     }
+
+    const existingPixel = props.extraPixelsData.find(
+      (pixel) =>
+        pixel.x == props.selectedPositionX && pixel.y == props.selectedPositionY
+    );
+
+    if (existingPixel) {
+      props.setSelectedColorId(existingPixel.colorId);
```
Comment (b-j-roberts) [inline frontend/src/canvas/CanvasContainer.js:269]: We don't want to change the selected color id here, but use the color in the extra pixels data to return the inverse like this 
```
return (
        '#' +
        (255 - color[0]).toString(16).padStart(2, '0') +
        (255 - color[1]).toString(16).padStart(2, '0') +
        (255 - color[2]).toString(16).padStart(2, '0') +
        color[3].toString(16).padStart(2, '0')
      );```

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: frontend/src/App.js (hunk 1 above)
Comment (kateberryd) [reply frontend/src/App.js:132]: My bad. I pushed test data

File: frontend/src/canvas/CanvasContainer.js [hunk 5]
```diff
@@ -259,8 +259,24 @@ const CanvasContainer = (props) => {
...
+    const existingPixel = props.extraPixelsData.find(
+      (pixel) =>
+        pixel.x == props.selectedPositionX && pixel.y == props.selectedPositionY
+    );
+
+    if (existingPixel) {
+      let color = props.colors[existingPixel.colorId];
+      return (
+        '#' +
+        (255 - parseInt(color[0], 16)).toString(16).padStart(2, '0') +
```
Comment (b-j-roberts) [inline frontend/src/canvas/CanvasContainer.js:272]: colors are formatted like "RRGGBB", so for example "FA0000" color[0]="F" and color[1]="A". Both need to be used to parse the red value.

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

**Comment by b-j-roberts** [comment]
Thank you! Nice work!

**Pull request merged**